scripts/jina-hub-update.py

```diff
@@ -0,0 +1,77 @@
+""" Script to change versioning of files (eg. manifest.yml) for executors
+[encoders, crafters, indexers, rankers].
+It also adds the required jina version.
+Commits the change in the branch and raises a PR for the executor.
+"""
+import glob
+import os
+import subprocess
+import sys
+
+import git
+from ruamel.yaml import YAML
+
+repo_dir = os.getcwd()
+yaml = YAML()
+
+version_file = 'manifest.yml'
+
+
+def main():
+    hub_repo = git.Repo('jina/hub')
+    hub_origin = hub_repo.remote(name='origin')
+    hub_origin_url = list(hub_origin.urls)[0]
+    assert 'jina-hub.git' in hub_origin_url, f'hub repo was not initialized correctly'
+
+    jina_core_repo = git.Repo('.')
+    core_origin_url = list(jina_core_repo.remote(name='origin').urls)[0]
+    assert 'jina.git' in core_origin_url, f'core repo was not initialized correctly'
+
+    jina_core_version = jina_core_repo.tags[-1].tag.tag[1:]  # remove leading 'v'
+    previous_branch = hub_repo.active_branch
+
+    print(f'cur. dir. is "{os.getcwd()}"')
+    print(f'got jina core v: "{jina_core_version}"')
+
+    # TODO remove limit set during dev.
+    for fpath in glob.glob(f'./jina/hub/**/{version_file}', recursive=True)[:1]:
```

Comment:
so this recursion would need to be done  inside the cloned `jina-hub` repo I guess or whatever API u cna use

Reply:
Yep, updated and it seems to work. There are still some problems with conflicting branch names (if there is one that wasn't merged yet) and still need to figure out how to properly set up auth in GH actions